kentikapi/internal/api_payloads/saved_filter.go

```diff
@@ -0,0 +1,182 @@
+package api_payloads
+
+import (
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/utils"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+)
+
+type GetAllSavedFilterResponse []GetSavedFilterResponse
+
+func (p GetAllSavedFilterResponse) ToSavedFilters() ([]models.SavedFilter, error) {
+	var result []models.SavedFilter
+	err := utils.ConvertList(p, GetSavedFilterResponse.ToSavedFilter, &result)
+	if err != nil {
+		return nil, err
+	}
+	return result, err
+}
+
+type GetSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type CreateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterRequest struct {
+	SavedFilterPayload
+}
+
+type UpdateSavedFilterResponse struct {
+	SavedFilterPayload
+}
+
+type SavedFilterPayload struct {
+	ID                IntAsString    `json:"id,omitempty" response:"get,post,put"`
+	CompanyID         models.ID      `json:"company_id,string,omitempty" response:"get,post,put"`
+	FilterName        string         `json:"filter_name" response:"get,post,put"`
+	FilterDescription string         `json:"filter_description" response:"get,post,put"`
+	FilterLevel       string         `json:"filter_level,omitempty" response:"get,post,put"`
+	CreatedDate       *time.Time     `json:"cdate,omitempty" response:"get,post,put"`
+	UpdatedDate       *time.Time     `json:"edate,omitempty" response:"get,post,put"`
+	Filters           FiltersPayload `json:"filters" response:"get,post,put"`
+}
+
+func (p SavedFilterPayload) ToSavedFilter() (models.SavedFilter, error) {
+	filters, err := p.Filters.ToFilters()
+	if err != nil {
+		return models.SavedFilter{}, err
+	}
+
+	return models.SavedFilter{
+		ID:                int(p.ID),
+		CompanyID:         p.CompanyID,
+		FilterName:        p.FilterName,
+		FilterDescription: p.FilterDescription,
+		FilterLevel:       p.FilterLevel,
+		CreatedDate:       *p.CreatedDate,
+		UpdatedDate:       *p.UpdatedDate,
+		Filters:           filters,
+	}, nil
+}
+
+func SavedFilterToPayload(sf models.SavedFilter) SavedFilterPayload {
+	return SavedFilterPayload{
+		ID:                IntAsString(sf.ID),
+		CompanyID:         sf.CompanyID,
+		FilterName:        sf.FilterName,
+		FilterDescription: sf.FilterDescription,
+		FilterLevel:       sf.FilterLevel,
+		CreatedDate:       &sf.CreatedDate,
+		UpdatedDate:       &sf.UpdatedDate,
+		Filters:           FiltersToPayload(sf.Filters),
+	}
+}
+
+type FiltersPayload struct {
```

Comment:
I think that in this payload no request/response validation tags make sense. See explanation above.

Reply:
Added a follow-up question